plugins/notifiers/slack/client.go

```diff
@@ -63,26 +75,45 @@ func NewNotifier(config *Config, httpClient utils.HTTPClient) *Notifier {
 func (n *Notifier) Notify(items []domain.Notification) []error {
 	errs := make([]error, 0)
 	for _, item := range items {
+		var ws SlackWorkspace
+		var slackID string
 		labelSlice := utils.MapToSlice(item.Labels)
-		slackID, err := n.findSlackIDByEmail(item.User)
-		if err != nil {
-			errs = append(errs, fmt.Errorf("%v | %w", labelSlice, err))
+
+		// check cache
+		if n.slackIDCache[item.User] != nil {
+			slackID = n.slackIDCache[item.User].SlackID
+			ws = n.slackIDCache[item.User].Workspace
+		} else {
+			ws, err := n.getSlackWorkspaceForUser(item.User)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("%v | %w", labelSlice, err))
+			}
+			slackID, err = n.findSlackIDByEmail(item.User, ws)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("%v | %w", labelSlice, err))
+			}
+
+			// cache
+			n.slackIDCache[item.User] = &slackIDCacheItem{
```

Comment:
Same here, we can skip updating cache if previous function calls return error

Reply:
updated.